### ISSUE TITLE
A forwarded message does not contain an ID in the message context

### ISSUE THREAD

**Issue by angelomelonas** [netflie/whatsapp-cloud-api#94]
First of all, thank you for this excellent library!

I would like to report a bug that I found. If a voice note (media message) is forwarded, it does not contain the `id` field in `$message['context']['id']`, resulting in the following exception:
```json
{
  "message": "Uncaught PHP Exception TypeError: \"Netflie\\WhatsAppCloudApi\\WebHook\\Notification\\Support\\Context::__construct(): Argument #1 ($replying_to_message_id) must be of type string, null given, called in /app/vendor/netflie/whatsapp-cloud-api/src/WebHook/Notification/MessageNotificationFactory.php on line 131\" at /app/vendor/netflie/whatsapp-cloud-api/src/WebHook/Notification/Support/Context.php line 13",
  "context": {
    "exception": {
      "class": "TypeError",
      "message": "Netflie\\WhatsAppCloudApi\\WebHook\\Notification\\Support\\Context::__construct(): Argument #1 ($replying_to_message_id) must be of type string, null given, called in /app/vendor/netflie/whatsapp-cloud-api/src/WebHook/Notification/MessageNotificationFactory.php on line 131",
      "code": 0,
      "file": "/app/vendor/netflie/whatsapp-cloud-api/src/WebHook/Notification/Support/Context.php:13"
    }
  }
}
```

Would a solution be to make the `id` field in the `Context` object optional? Then the code would be:

```php
$notification->withContext(new Support\Context(
    $message['context']['id'] ?? null,
    $message['context']['forwarded'] ?? false,
    $referred_product ?? null
));
```

I'd be happy to [open a pull request](https://github.com/netflie/whatsapp-cloud-api/compare/main...angelomelonas:whatsapp-cloud-api:main) for this 👍🏼

**Comment by aalbarca** [comment]
I think maybe the problem could be that the field is `message_id` instead of `id`, could you send me a trace of the raw response sent by Facebook?

Thanks!

**Comment by aalbarca** [comment]
Any news @angelomelonas ?

**Comment by angelomelonas** [comment]
Hi @aalbarca! My apologies, I was recently moving, so I was not able to have another look at this. I'll get back to you ASAP.

**Comment by angelomelonas** [comment]
It does indeed seem like the `id`/`message_id` field(s) are missing from the context when a message (text/media) is forwarded. The following is an example of a voice message that was forwarded:

```php
array:2 [
  "object" => "whatsapp_business_account"
  "entry" => array:1 [
    0 => array:2 [
      "id" => "101194536240516"
      "changes" => array:1 [
        0 => array:2 [
          "value" => array:4 [
            "messaging_product" => "whatsapp"
            "metadata" => array:2 [
              "display_phone_number" => "..."
              "phone_number_id" => "101573099534713"
            ]
            "contacts" => array:1 [
              0 => array:2 [
                "profile" => array:1 [
                  "name" => "..."
                ]
                "wa_id" => "..."
              ]
            ]
            "messages" => array:1 [
              0 => array:5 [
                "from" => "..."
                "id" => "..."
                "timestamp" => "1679567269"
                "type" => "audio"
                "audio" => array:4 [
                  "mime_type" => "audio/ogg; codecs=opus"
                  "sha256" => "+NH2CUzuKuNRz73+hEp0U/GQaC/IzUEP5IcznZ6/C4E="
                  "id" => "1218877259015941"
                  "voice" => true
                ]
              ]
            ]
          ]
          "field" => "messages"
        ]
      ]
    ]
  ]
]
```

**Comment by angelomelonas** [comment]
Have you perhaps had a moment to look the response, @aalbarca? 😄

**Comment by aalbarca** [comment]
Could you maybe open a PR with the fix? 🙏

**Comment by angelomelonas** [comment]
@aalbarca [pull request #100](https://github.com/netflie/whatsapp-cloud-api/pull/100) 😄

**Comment by aalbarca** [comment]
Fixed on 2.0.3 release. Thanks @angelomelonas for the PR ;)
